### PR TITLE
support array-like objects in R.slice

### LINKED ISSUE #839: R.init(arguments) => TypeError: Cannot read property 'apply' of undefined
Is this a known issue?

[Ramda REPL](http://goo.gl/hyAdQ7)

### PR DESCRIPTION
**Pull request by davidchambers**
Fixes #839

### CODE REVIEW

**Comment by megawac** [comment]
The flip side is this will break on objects with custom slices

**Comment by davidchambers** [comment]
> The flip side is this will break on objects with custom slices

True. Do you think we should special-case Arguments objects?

**Comment by davidchambers** [comment]
I updated the pull request to use `_checkForMethod`, and added a test case for an object with a custom `slice` method.

File: dist/ramda.js [hunk 1]
```diff
@@ -4420,22 +4420,25 @@
...
-     * @return {Array} The items from `a` to `b - 1` from `xs`.
+     * @return {Array} The slice of `xs` from `fromIndex` to `toIndex`.
      * @example
      *
      *      var xs = R.range(0, 10);
      *      R.slice(2, 5)(xs); //=> [2, 3, 4]
      */
-    var slice = invoker(2, 'slice');
+    var slice = _curry3(_checkForMethod('slice', function slice(fromIndex, toIndex, xs) {
+        return Array.prototype.slice.call(xs, fromIndex, toIndex);
```
Comment (buzzdecafe) [inline dist/ramda.js]: why not call internal `_slice`?

File: test/init.js [hunk 2]
```diff
@@ -15,4 +15,8 @@ describe('init', function() {
         assert.throws(function() { R.init(undefined); }, TypeError);
         assert.throws(function() { R.init(); }, TypeError);
     });
+    it('handles array-like object', function() {
+        var args = (function() { return arguments; }(1, 2, 3, 4, 5));
+        assert.deepEqual(R.init(args), [1, 2, 3, 4]);
+    });
```
Comment (buzzdecafe) [inline test/init.js]: important to note that unless the array-like object has its own `slice` method, `R.slice` *always* returns an Array, not a "sliced" version of the input array-like object.

File: dist/ramda.js (hunk 1 above)
Comment (CrossEye) [inline dist/ramda.js]: Internal slice only handles positive values.

File: src/slice.js [hunk 3]
```diff
@@ -1,20 +1,24 @@
-var invoker = require('./invoker');
+var _checkForMethod = require('./internal/_checkForMethod');
+var _curry3 = require('./internal/_curry3');
 
 
 /**
- * Returns the elements from `xs` starting at `a` and ending at `b - 1`.
+ * Returns the elements of `xs` from `fromIndex` (inclusive) to
```
Comment (CrossEye) [inline src/slice.js]: Perhaps

* Returns a list containing the elements from...

**Comment by CrossEye** [comment]
LGTM

File: src/slice.js (hunk 3 above)
Comment (davidchambers) [inline src/slice.js]: Yes, that's better. Updated.

**Pull request merged**